backend/src/apiserver/template/v2_template.go

```diff
@@ -73,6 +74,10 @@ func NewV2SpecTemplate(template []byte) (*V2Spec, error) {
 	if spec.GetPipelineInfo().GetName() == "" {
 		return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: name is empty")
 	}
+	match, _ := regexp.MatchString("[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*", spec.GetPipelineInfo().GetName())
```

Comment:
I think it's better to be more strict:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#path-segment-names

Because we also use pipeline names in paths.

Besides, we may want to enforce a length limit as well.

Reply:
I seem to be able to run a pipeline with `.` inside the name:

![pipelinewithdot](https://user-images.githubusercontent.com/37026441/144661321-59120df1-a0e2-410f-a772-41aefc23a65a.png)

And the pipeline length is currently limited to 100 characters:

![pipelinetoolong](https://user-images.githubusercontent.com/37026441/144661365-03f26e1f-cc94-4be7-b45d-f379e6bf9f0f.png)

Would you like to share the place where we use pipeline names in paths? Thank you!

Reply:
Yes, we append pipeline name and run ID to pipeline root

Reply:
https://github.com/kubeflow/pipelines/blob/fef8c03e401a15a9f92c1839fe0f9a5c22f709e1/v2/metadata/client.go#L250

We can also log an issue to discuss this later.

Reply:
Thank you for the reference! Discussed offline and the subdomain rule is acceptable by path segment rule.